In [1]:
import pandas as pd
import datetime as dt
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn import model_selection
from sklearn.model_selection import train_test_split
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import mean_squared_error
fie = pd.read_csv("../Family Income and Expenditure.csv")

## 1. รายชื่อกลุ่ม 

- นายกลวัชร เสถียรเสรีสุข  6210450016
- นายเกียรติกุล ทองอุดม    6210450032
- นานชาญวิทย์ สายสิญจน์  6210450521
- น.ส.ณัฎฐกานต์ อรสุนทร   6210451144
- น.ส.ธันวพร ธรรมเจริญพงศ์ 6210451225

## 2. รายละเอียดข้อมูล และการทำความสะอาดข้อมูล ##

**คำอธิบายข้อมูล**
- เป็นการเก็บข้อมูลที่ได้จากการสำรวจรายได้และรายจ่ายของแต่ละครอบครัว ในฟิลิปปินส์ มีประมาณ 40,000 กว่าครัวเรือน โดยเก็บข้อมูลทุกๆ 3 ปี

**แหล่งที่มาของข้อมูล**
- https://www.kaggle.com/grosvenpaul/family-income-and-expenditure

**เก็บรวบรวมโดย**
- Francis Paul Flores

**รายละเอียดคอลัมน์(สนใจแค่คอลัมน์ที่เกี่ยวกับรายรับรายจ่าย และข้อมูลของครอบครัว)**
1. Total Household Income - รายได้ทั้งหมดของแต่ละครัวเรือน type(int)
2. Region - ภูมิภาคของแต่ละครัวเรือน type(object)
3. Total Food Expenditure - รายจ่ายทั้งหมดที่ใช้กับอาหาร type(int)
4. Main Source of Income - แหล่งรายได้หลัก type(object)
5. Agricultural Household indicator - ตัวบ่งชี้ครัวเรือนเกษตร type(int)
6. Bread and Cereals Expenditure - รายจ่ายที่ใช้กับขนมปังและธัญพืช type(int)
7. Total Rice Expenditure - รายจ่ายทั้งหมดที่ใช้กับข้าวtype(int)
8. Meat Expenditure - รายจ่ายทั้งหมดที่ใช้กับเนื้อ type(int)
9. Total Fish and marine products Expenditure - รายจ่ายทั้งหมดที่ใช้กับปลาและภลิตภัณฑ์ที่ทำจากทะเล type(int)
10. Fruit Expenditure - รายจ่ายที่ใช้กับผลไม้ type(int)
11. Vegetables Expenditure - รายจ่ายที่ใช้กับผัก type(int)
12. Restaurant and hotels Expenditure - รายจ่ายที่ใช้กับร้านอาหารและโรงแรม type(int)
13. Alcoholic Beverages Expenditure - รายจ่ายที่ใช้กับเครื่องดื่มแอลกอฮอล์ type(int)
14. Tobacco Expenditure - ค่าใช้จ่ายที่ใช้กับยาสูบ type(int)
15. Clothing, Footwear and Other Wear Expenditure - รายจ่ายที่ใช้กับเสื้อผ้า รองเท้า และเครื่องแต่งกายอื่นๆ type(int)
16. Housing and water Expenditure - ค่าใช้จ่ายที่เกี่ยวกับค่าน้ำและค่าไฟ type(int)
17. Imputed House Rental Value - ค่าเช่าบ้าน type(int)
18. Medical Care Expenditure - รายจ่ายที่ใช้กับค่ารักษาพยาบาล type(int)
19. Transportation Expenditure - รายจ่ายที่ใช้กับการเดินทาง type(int)
20. Communication Expenditure - ค่าใช้จ่ายในกาสื่อสาร type(int)
21. Education Expenditure - รายจ่ายที่ใช้กับการษึกษาเล่าเรียน type(int)
22. Miscellaneous Goods and Services Expenditure - รายจ่ายที่ใช้กับสินค้าและบริการเบ็ดเตล็ด type(int)
23. Special Occasions Expenditure - รายจ่ายที่ใช้ในโอกาสพิเศษ type(int)
24. Crop Farming and Gardening expenses - รายจ่ายที่ใช้กับการปลูกพืชและการทำสวน type(int)
25. Total Income from Entrepreneurial Acitivites - รายได้รวมจากกิจกรรมผู้ประกอบการ type(int)
26. Household Head Sex - เพศของหัวหน้าครั้วเรือน type(object)
27. Household Head Age - อายุของหัวหน้าครัวเรือน type(int)
28. Household Head Marital Status - สถานภาพการสมรสของหัวหน้าครัวเรือน type(object)
29. Household Head Highest Grade Completed - การจบการศึกษาของหัวหน้าครัวเรือน type(object)
30. Household Head Job or Business Indicator - งานหรือธุรกิจของหัวหน้าครัวเรือน type(object)
31. Household Head Occupation - อาชัพของหัวหน้าครอบครัว type(object)
32. Household Head Class of Worker	type(object)
33. Type of Household - ประเภทของครัวเรือน type(object)
34. Total Number of Family members -  จำนวนสมาชิกของแต่ละครัวเรือน type(int)
35. Members with age less than 5 year old - 
36. Members with age 5 - 17 years old - 
37. Total number of family members employed - 
38. Type of Building/House - 
39. Type of Roof - 
40. Type of Walls - 
41. House Floor Area - 
42. House Age - 
43. Number of bedrooms - 
44. Tenure Status - 
45. Toilet Facilities - 
46. Electricity - 
47. Main Source of Water Supply -
48. Number of Television - 
49. Number of CD/VCD/DVD - 
50. Number of Component/Stereo set - 
51. Number of Refrigerator/Freezer - 
52. Number of Washing Machine - 
53. Number of Airconditioner - 
54. Number of Car, Jeep, Van - 
55. Number of Landline/wireless telephones - 
56. Number of Cellular phone - 
57. Number of Personal Computer - 
58. Number of Stove with Oven/Gas Range - 
59. Number of Motorized Banca - 
60. Number of Motorcycle/Tricycle - 

In [2]:
fie

Total Household Income              Region  Total Food Expenditure  \
0                      480332                 CAR                  117848   
1                      198235                 CAR                   67766   
2                       82785                 CAR                   61609   
3                      107589                 CAR                   78189   
4                      189322                 CAR                   94625   
...                       ...                 ...                     ...   
41539                  119773  XII - SOCCSKSARGEN                   44875   
41540                  137320  XII - SOCCSKSARGEN                   31157   
41541                  133171  XII - SOCCSKSARGEN                   45882   
41542                  129500  XII - SOCCSKSARGEN                   81416   
41543                  128598  XII - SOCCSKSARGEN                   78195   

            Main Source of Income  Agricultural Household indicator  \
0                   Wage/Salaries                                 0   
1                   Wage/Salaries                                 0   
2                   Wage/Salaries                                 1   
3                   Wage/Salaries                                 0   
4                   Wage/Salaries                                 0   
...                           ...                               ...   
41539  Enterpreneurial Activities                                 1   
41540  Enterpreneurial Activities                                 1   
41541  Enterpreneurial Activities                                 2   
41542  Enterpreneurial Activities                                 1   
41543  Enterpreneurial Activities                                 1   

       Bread and Cereals Expenditure  Total Rice Expenditure  \
0                              42140                   38300   
1                              17329                   13008   
2                              34182                   32001   
3                              34030                   28659   
4                              34820                   30167   
...                              ...                     ...   
41539                          23675                   21542   
41540                           2691                    1273   
41541                          28646                   27339   
41542                          29996                   26655   
41543                          43485                   41205   

       Meat Expenditure  Total Fish and  marine products Expenditure  \
0                 24676                                        16806   
1                 17434                                        11073   
2                  7783                                         2590   
3                 10914                                        10812   
4                 18391                                        11309   
...                 ...                                          ...   
41539              1476                                         6120   
41540              1886                                         4386   
41541               480                                         4796   
41542              2359                                        17730   
41543              1985                                         7735   

       Fruit Expenditure  ...  Number of Refrigerator/Freezer  \
0                   3325  ...                               1   
1                   2035  ...                               0   
2                   1730  ...                               0   
3                    690  ...                               0   
4                   1395  ...                               1   
...                  ...  ...                             ...   
41539               1632  ...                               0   
41540               1840  ...                               0   
41541     

In [3]:
fie.columns

Index(['Total Household Income', 'Region', 'Total Food Expenditure',
       'Main Source of Income', 'Agricultural Household indicator',
       'Bread and Cereals Expenditure', 'Total Rice Expenditure',
       'Meat Expenditure', 'Total Fish and  marine products Expenditure',
       'Fruit Expenditure', 'Vegetables Expenditure',
       'Restaurant and hotels Expenditure', 'Alcoholic Beverages Expenditure',
       'Tobacco Expenditure', 'Clothing, Footwear and Other Wear Expenditure',
       'Housing and water Expenditure', 'Imputed House Rental Value',
       'Medical Care Expenditure', 'Transportation Expenditure',
       'Communication Expenditure', 'Education Expenditure',
       'Miscellaneous Goods and Services Expenditure',
       'Special Occasions Expenditure', 'Crop Farming and Gardening expenses',
       'Total Income from Entrepreneurial Acitivites', 'Household Head Sex',
       'Household Head Age', 'Household Head Marital Status',
       'Household Head Highest Grade Compl

### ปรับเปลี่ยนค่า null 

In [4]:
fie.loc[fie["Household Head Occupation"].isnull(),["Household Head Occupation"]] ='Unknown'
fie.loc[fie["Household Head Class of Worker"].isnull(),["Household Head Class of Worker"]] = 'Unknown'
fie

Total Household Income              Region  Total Food Expenditure  \
0                      480332                 CAR                  117848   
1                      198235                 CAR                   67766   
2                       82785                 CAR                   61609   
3                      107589                 CAR                   78189   
4                      189322                 CAR                   94625   
...                       ...                 ...                     ...   
41539                  119773  XII - SOCCSKSARGEN                   44875   
41540                  137320  XII - SOCCSKSARGEN                   31157   
41541                  133171  XII - SOCCSKSARGEN                   45882   
41542                  129500  XII - SOCCSKSARGEN                   81416   
41543                  128598  XII - SOCCSKSARGEN                   78195   

            Main Source of Income  Agricultural Household indicator  \
0                   Wage/Salaries                                 0   
1                   Wage/Salaries                                 0   
2                   Wage/Salaries                                 1   
3                   Wage/Salaries                                 0   
4                   Wage/Salaries                                 0   
...                           ...                               ...   
41539  Enterpreneurial Activities                                 1   
41540  Enterpreneurial Activities                                 1   
41541  Enterpreneurial Activities                                 2   
41542  Enterpreneurial Activities                                 1   
41543  Enterpreneurial Activities                                 1   

       Bread and Cereals Expenditure  Total Rice Expenditure  \
0                              42140                   38300   
1                              17329                   13008   
2                              34182                   32001   
3                              34030                   28659   
4                              34820                   30167   
...                              ...                     ...   
41539                          23675                   21542   
41540                           2691                    1273   
41541                          28646                   27339   
41542                          29996                   26655   
41543                          43485                   41205   

       Meat Expenditure  Total Fish and  marine products Expenditure  \
0                 24676                                        16806   
1                 17434                                        11073   
2                  7783                                         2590   
3                 10914                                        10812   
4                 18391                                        11309   
...                 ...                                          ...   
41539              1476                                         6120   
41540              1886                                         4386   
41541               480                                         4796   
41542              2359                                        17730   
41543              1985                                         7735   

       Fruit Expenditure  ...  Number of Refrigerator/Freezer  \
0                   3325  ...                               1   
1                   2035  ...                               0   
2                   1730  ...                               0   
3                    690  ...                               0   
4                   1395  ...                               1   
...                  ...  ...                             ...   
41539               1632  ...                               0   
41540               1840  ...                               0   
41541     

In [5]:
expenditure_col= []
for i in fie.columns:
    if ("Expenditure" in i) :
        expenditure_col.append(i)
expenditure_col

['Total Food Expenditure',
 'Bread and Cereals Expenditure',
 'Total Rice Expenditure',
 'Meat Expenditure',
 'Total Fish and  marine products Expenditure',
 'Fruit Expenditure',
 'Vegetables Expenditure',
 'Restaurant and hotels Expenditure',
 'Alcoholic Beverages Expenditure',
 'Tobacco Expenditure',
 'Clothing, Footwear and Other Wear Expenditure',
 'Housing and water Expenditure',
 'Medical Care Expenditure',
 'Transportation Expenditure',
 'Communication Expenditure',
 'Education Expenditure',
 'Miscellaneous Goods and Services Expenditure',
 'Special Occasions Expenditure']

## 3. กระบวนการ Machine learning
### โจทย์การวิเคราะห์เชิงลึก - ในแต่ละครอบครัวมีการใช้จ่ายค่าอาหารและค่าวัตถุดิบอย่างไร

### จุดมุ่งหมายของโมเดลทั้งหมด
- เพื่อทำนายค่าใช้จ่ายในด้านของอาหารและวัตถุดิบในการทำอาหารจากปัจจัยต่างๆของแต่ละครัวเรือน
- นำมาประกอบการตัดสินใจหรือวางแผนในการใช้เงินในอนาคตได้
- เพื่อเป็นแนวทางแต่ละครัวเรือนมีความจำเป็นในวัตถุดิบประเภทไหนมาก อาจนำมาใช้ในการลงทุนได้
- เพื่อทดสอบความสัมพันธ์ของคอลัมป์ที่เป็นค่าใช้จ่ายในด้านอาหารและวัตถุดิบ ขึ้นอยู่กับคอลัมป์ใดบ้าง

#### เลือกตัวแปรต้นที่อาจส่งผลต่อค่าอาหารและวัตถุดิบทั้งหมด
- Total Household Income
- Total Number of Family members
- Members with age less than 5 year old
- Members with age 5 - 17 years old
- Total number of family members employed
- Type of Household

### 3.1 สร้างโมเดลเพื่อทำนายค่าอาหาร(Total Food Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [6]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Total Food Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [7]:
tf = LinearRegression()
tf.fit(X_train, y_train)

LinearRegression()

In [8]:
tf.coef_, tf.intercept_

(array([ 1.02134320e-01,  8.43205674e+03, -4.69671330e+03, -2.55160624e+03,
         3.53357144e+03, -1.08760144e+03, -3.25784494e+03,  4.34544638e+03]),
 24108.32792758903)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [9]:
tf.score(X_test, y_test)

0.5913002386127275

In [10]:
print('Coefficient of determination = %f%%' %(tf.score(X_test, y_test)*100))

Coefficient of determination = 59.130024%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าอาหาร เท่ากับ 59.13% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 59.13%

### 3.2 สร้างโมเดลเพื่อทำนายค่าวัตถุดิบจำพวกขนมปัง(Bread and Cereals Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [11]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Bread and Cereals Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [12]:
bce = LinearRegression()
bce.fit(X_train, y_train)

LinearRegression()

In [13]:
bce.coef_, bce.intercept_

(array([ 8.17569284e-03,  3.83124530e+03, -2.00474944e+03, -2.53912317e+02,
         1.97552497e+02, -7.43119296e+02, -7.24447932e+02,  1.46756723e+03]),
 7004.730745056666)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [14]:
bce.score(X_test, y_test)

0.45834706389474866

In [15]:
print('Coefficient of determination = %f%%' %(bce.score(X_test, y_test)*100))

Coefficient of determination = 45.834706%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปัง เท่ากับ 45.83% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 45.83%

### 3.3 สร้างโมเดลเพื่อทำนายค่าข้าว(Total Rice Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [16]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Total Rice Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [17]:
tre = LinearRegression()
tre.fit(X_train, y_train)

LinearRegression()

In [18]:
tre.coef_, tre.intercept_

(array([ 2.90825240e-03,  3.14706837e+03, -1.87033416e+03, -2.95584794e+02,
         1.49313863e+02, -5.00268415e+02, -3.30581559e+02,  8.30849974e+02]),
 4275.103902200848)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [19]:
tre.score(X_test, y_test)

0.35890868436755585

In [20]:
print('Coefficient of determination = %f%%' %(tre.score(X_test, y_test)*100))

Coefficient of determination = 35.890868%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปัง เท่ากับ 35.89% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 35.89%

### 3.4 สร้างโมเดลเพื่อทำนายค่าวัตถุดิบจำพวกเนื้อสัตว์(Meat Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [21]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Meat Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [22]:
me = LinearRegression()
me.fit(X_train, y_train)

LinearRegression()

In [23]:
me.coef_, me.intercept_

(array([ 1.83485802e-02,  7.09605304e+02, -8.15636930e+02, -3.95038188e+02,
         7.66659602e+02,  2.50037971e+02, -9.68400617e+02,  7.18362646e+02]),
 3167.6013113293366)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [24]:
me.score(X_test, y_test)

0.3588539226881763

In [25]:
print('Coefficient of determination = %f%%' %(me.score(X_test, y_test)*100))

Coefficient of determination = 35.885392%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปัง เท่ากับ 35.88% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 35.88%

### 3.5 สร้างโมเดลเพื่อทำนายค่าวัตถุดิบทางทะเล(Total Fish and  marine products Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [26]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Total Fish and  marine products Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [27]:
tfmpe = LinearRegression()
tfmpe.fit(X_train, y_train)

LinearRegression()

In [28]:
tfmpe.coef_, tfmpe.intercept_

(array([ 9.13270367e-03,  1.45081105e+03, -1.25729438e+03, -6.88288609e+02,
        -2.90382434e+02, -1.26268626e+02,  6.99771901e+01,  5.62914357e+01]),
 3358.9642572068897)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [29]:
tfmpe.score(X_test, y_test)

0.22527366289574402

In [30]:
print('Coefficient of determination = %f%%' %(tfmpe.score(X_test, y_test)*100))

Coefficient of determination = 22.527366%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปัง เท่ากับ 22.52% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 22.52%

### 3.6 สร้างโมเดลเพื่อทำนายค่าวัตถุดิบจำพวกผลไม้(Fruit Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [31]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Fruit Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [32]:
fe = LinearRegression()
fe.fit(X_train, y_train)

LinearRegression()

In [33]:
fe.coef_, fe.intercept_

(array([ 4.44924093e-03,  1.57461381e+02, -1.80762237e+02, -1.18059292e+02,
        -1.09094556e+02, -2.37326248e+01, -1.16529121e+02,  1.40261746e+02]),
 1178.335194909068)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [34]:
fe.score(X_test, y_test)

0.25245802052785093

In [35]:
print('Coefficient of determination = %f%%' %(fe.score(X_test, y_test)*100))

Coefficient of determination = 25.245802%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปัง เท่ากับ 25.24% กล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 25.24%

### 3.7 สร้างโมเดลเพื่อทำนายค่าวัตถุดิบจำพวกผัก(Vegetables Expenditure) โดยใช้ Regression

#### Splitting data into training and testing sets - สร้างชุดทดลองและชุดทดสอบ

In [36]:
X = pd.get_dummies(fie.loc[:,['Total Household Income','Total Number of Family members','Members with age less than 5 year old',
               'Members with age 5 - 17 years old','Total number of family members employed','Type of Household']])
y = fie["Vegetables Expenditure"]
X_train, X_test, y_train, y_test = train_test_split(X, y)

#### เรียกใช้ Linear Regression แล้วนำข้อมูลที่ split ไว้มา fit

In [37]:
ve = LinearRegression()
ve.fit(X_train, y_train)

LinearRegression()

In [38]:
ve.coef_, ve.intercept_

(array([ 3.52069795e-03,  5.12885998e+02, -4.74264629e+02, -2.12756203e+02,
        -5.70965450e+01, -2.68829102e+02, -3.52512269e+02,  6.21341371e+02]),
 2641.1840381851107)

#### วัดประสิทธิภาพของโมเดลการทำนายโดยคำนวณค่า $R^2$ จากฟังก์ชัน score()

In [39]:
ve.score(X_test, y_test)

0.2151818291660652

In [40]:
print('Coefficient of determination = %f%%' %(ve.score(X_test, y_test)*100))

Coefficient of determination = 21.518183%


จะได้ว่าความแปรปรวนที่ลดลงไปเมื่อใช้โมเดลนี้ทำนายค่าวัตถุดิบจำพวกขนมปังประมาณ 21.51% หรือกล่าวได้ว่าโมเดลการทำนายมีประสิทธิภาพ 21.51%

#### จากกระบวน Machine learning แบบ Regression ข้างต้น จะได้โมเดลในการหาค่าอาหารและวัตถุดิบมาทั้งหมด 7 โมเดลด้วยกัน จากนั้นจะต้องนำไปคำนวณโดยการรวมผลของการทำนายที่ได้เข้าด้วยกัน เราจะได้ค่าใช้จ่ายในด้านอาหารและวัตถุดิบในการทำอาหารทั้งหมด

## 4. อธิบายสรุปผล

#### ข้อมูลที่จำเป็นสำหรับการคำนวณและทำนายในโมเดลข้างต้นในข้อ 3 มีดังนี้
- Total Household Income
- Total Number of Family members
- Members with age less than 5 year old
- Members with age 5 - 17 years old
- Total number of family members employed
- Type of Household

#### แต่ละโมเดลทำนายอะไร
- 
- 
- 
- 

#### ประโยชน์ของโมเดลและการนำโมเดลไปใช้
- สามารถนำไปประกอบการวางแผนการใช้จ่ายให้ดีหรือประหยัดขึ้นได้
-
-
-

#### วิเคราะห์โมเดล
- 
- 
- 
- 
- 
- 
- 